packages/console/src/pages/Callback/index.tsx

```diff
@@ -1,11 +1,21 @@
-import { useHandleSignInCallback } from '@logto/react';
+import { LogtoError, OidcError, useHandleSignInCallback } from '@logto/react';
 import React from 'react';
 
+import AppError from '@/components/AppError';
 import LogtoLoading from '@/components/LogtoLoading';
 import { getBasename } from '@/utilities/app';
 
 const Callback = () => {
-  useHandleSignInCallback(getBasename());
+  const { error } = useHandleSignInCallback(getBasename());
+
+  if (error) {
+    const errorMessage =
+      error instanceof LogtoError && error.data instanceof OidcError
+        ? error.data.errorDescription
+        : error.message;
+
+    return <AppError errorMessage={errorMessage} callStack={error.stack} />;
```

Comment:
nit: show error code as well?

Reply:
How about we only use `error` code when `errorMessage` is empty?

Reply:
After short discussion, we'll add error code and the format would be `${errorCode}: ${errorMessage}`